src/chain.cpp

```diff
@@ -20,6 +20,20 @@ void CChain::SetTip(CBlockIndex *pindex) {
     }
 }
 
+void CChain::FakeNTx(unsigned int nChainTx) {
+    for (CBlockIndex* index : vChain) {
+        if (!index->nTx) {
+            index->nTx = 1;
+        }
+        index->nChainTx = index->pprev ? index->pprev->nChainTx + index->nTx : 1;
```

Comment:
In commit "chain: add unused CChain::FakeNTx" (ebc844e8a1bacfe603db85ef4deae6e1f109cd57)

It seems really unfortunate that this is necessary just to report progress. Are there alternative ways to change the progress function so this would not be required?